test_scripts/Defects/5_1/1393_ListFiles_from_enumerating_through_the_file_system.lua

```diff
@@ -0,0 +1,73 @@
+---------------------------------------------------------------------------------------------
+-- Issue: https://github.com/smartdevicelink/sdl_core/issues/1393
+--
+-- Description: Check that SDL responds to ListFiles request from enumerating through the file system
+--
+-- Precondition:
+-- 1) Core and HMI are started.
+-- 2) Application is registered
+-- 3) Upload the icon.png file
+-- 4) App sends the ListFiles request to the SDL and receives a response with the icon.png file from the SDL
+-- 5) App sends the DeleteFile request for the icon.png file to the SDL
+-- Step:
+-- 1) App sends the ListFiles request to the SDL
+-- SDL does:
+-- - send ListFiles response with an empty list to the App
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local common = require('user_modules/sequences/actions')
+local runner = require('user_modules/script_runner')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+local requestParams = {
+  syncFileName = "icon.png",
+  fileType = "GRAPHIC_PNG",
+  persistentFile = false,
+  systemFile = false
+}
+
+--[[ Local Functions ]]
+local function putFile()
+  local cid = common.getMobileSession():SendRPC("PutFile", requestParams ,"files/icon.png" )
+  common.getMobileSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+end
+
+local function listFiles(pResult)
+  local cid = common.getMobileSession():SendRPC("ListFiles", {} )
+  if pResult then
+    common.getMobileSession():ExpectResponse(cid,
+      { success = true, resultCode = "SUCCESS", filenames = { "icon.png" } })
+  else
+    common.getMobileSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+    :ValidIf (function(_,data)
+      if data.payload.filenames then
+        return false, "ListFiles response contains unexpected parameter 'filenames' \n"
+      end
+      return true
+    end)
+  end
+end
+
+local function deleteFile()
+  local cid = common.getMobileSession():SendRPC("DeleteFile", { syncFileName = requestParams.syncFileName })
+  common.getMobileSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
```

Comment:
This doesn't cover the original issue. The issue was that if the OS decided to delete a file in the background, the file would still appear in ListFiles. This should be replaced with an OS call to remove the file.